libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -656,6 +656,15 @@ void CompilerUtils::convertType(Type const& _typeOnStack, Type const& _targetTyp
 		solAssert(_typeOnStack == _targetType, "Invalid type conversion requested.");
 		break;
 	}
+
+	// Check the conversion result fits in a range.
+	if (targetTypeCategory == Type::Category::Enum)
```

Comment:
I think having this routine at this spot is generally confusing. We can keep an assert here which checks that only int to enum conversion is possible.